pkg/sources/filesystem/filesystem.go

```diff
@@ -259,8 +261,8 @@ func (s *Source) ChunkUnit(ctx context.Context, unit sources.SourceUnit, reporte
 		}
 	}
 
-	if scanErr != nil && scanErr != io.EOF {
-		logger.Info("error scanning filesystem", "error", scanErr)
+	if scanErr != nil && !errors.Is(scanErr, io.EOF) && !errors.Is(scanErr, skipSymlinkErr) {
+		logger.Error(scanErr, "error scanning filesystem")
 		return reporter.ChunkErr(ctx, scanErr)
 	}
```

Comment:
I think we should still keep the information in the report, but I agree that it doesn't need to be logged.

**Suggestion:**

```go
if scanErr != nil && !errors.Is(scannErr, io.EOF) {
    if !errors.Is(scanErr, skipSymlinkErr) {
        logger.Error(scanErr, "error scanning filesystem")
    }
    return reporter.ChunkErr(ctx, scanErr)
}
```

Reply:
> I think we should still keep the information in the report,

What are the implications of this? I don't really understand the function of `reporter`.

It seems to be that `skipSymlinkErr` is a sentinel value to say "skip this value", rather than an actual error. Elsewhere in the codebase, symlinks are skipped by returning `nil` rather than a special `err` case.

https://github.com/trufflesecurity/trufflehog/blob/f122b295bf4d80edf9218bef2a454a60c039be62/pkg/handlers/archive.go#L137-L139

Reply:
The goal of `reporter` is to track what is being scanned and any errors encountered at a more granular level.

Given that, I was thinking it would be good to see in the report "/tmp/foo/bar wasn't scanned because it's a symlink."

If you're interested in the reporting capability, there's a hidden flag [--output-report](https://github.com/trufflesecurity/trufflehog/blob/f122b295bf4d80edf9218bef2a454a60c039be62/main.go#L71) you can use to write the report to a file. It's still hidden because the output format isn't finalized and could change. It's also only useful for sources that [support unit scanning](https://github.com/trufflesecurity/trufflehog/blob/f122b295bf4d80edf9218bef2a454a60c039be62/pkg/sources/sources.go#L80-L85) (only a handful right now).

Reply:
I see. That makes sense.